Keep-alive client tries to send request after connection close

When the connection is closed due to a timeout, the keep-alive timer keeps running, resulting in a panic a few seconds after the connection is closed.

```
{"level":"info","timestamp":"2023-12-30T18:34:30-06:00","caller":"chainsync/chainsync.go:214","msg":"reconnecting to backbone.cardano-mainnet.iohk.io:3001 due to error: protocol error: keep-alive: timeout waiting on transition from protocol state Server"}
...
panic: send on closed channel

goroutine 5368 [running]:
github.com/blinklabs-io/gouroboros/protocol.(*Protocol).SendMessage(...)
        /home/agaffney/go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.69.0/protocol/protocol.go:174
github.com/blinklabs-io/gouroboros/protocol/keepalive.(*Client).sendKeepAlive(0xc0006ce900)
        /home/agaffney/go/pkg/mod/github.com/blinklabs-io/gouroboros@v0.69.0/protocol/keepalive/client.go:82 +0x65
created by time.goFunc
        /usr/local/go/src/time/sleep.go:176 +0x2d
```

--- a/protocol/keepalive/client.go
+++ b/protocol/keepalive/client.go
@@ -63,7 +63,11 @@ func NewClient(protoOptions protocol.ProtocolOptions, cfg *Config) *Client {
 		if c.timer != nil {
 			// Stop timer and drain channel
 			if ok := c.timer.Stop(); !ok {
-				<-c.timer.C
+				// Read item from channel, if available
+				select {
+				case <-c.timer.C:
+				default:
+				}
 			}
 		}
 	}()
@@ -82,11 +86,22 @@ func (c *Client) sendKeepAlive() {
 	if err := c.SendMessage(msg); err != nil {
 		c.SendError(err)
 	}
-	// Reschedule timer
+	// Schedule timer
 	c.startTimer()
 }
 
 func (c *Client) startTimer() {
+	// Stop any existing timer
+	if c.timer != nil {
+		if ok := c.timer.Stop(); !ok {
+			// Read item from channel, if available
+			select {
+			case <-c.timer.C:
+			default:
+			}
+		}
+	}
+	// Create new timer
 	c.timer = time.AfterFunc(c.config.Period, c.sendKeepAlive)
 }
 
@@ -107,10 +122,6 @@ func (c *Client) messageHandler(msg protocol.Message) error {
 
 func (c *Client) handleKeepAliveResponse(msgGeneric protocol.Message) error {
 	msg := msgGeneric.(*MsgKeepAliveResponse)
-	// Start the timer again if we had one previously
-	if c.timer != nil {
-		defer c.startTimer()
-	}
 	if c.config != nil && c.config.KeepAliveResponseFunc != nil {
 		// Call the user callback function
 		return c.config.KeepAliveResponseFunc(msg.Cookie)

